haystack/preprocessor/preprocessor.py

```diff
@@ -53,18 +53,68 @@ def __init__(
         self.split_overlap = split_overlap
         self.split_respect_sentence_boundary = split_respect_sentence_boundary
 
-    def clean(self, document: dict) -> dict:
+    def process(
+        self,
+        document: dict,
```

Comment:
I see why we want the option to overwrite instance attributes here, but it will create quite a mess if we apply this pattern to all potential pipeline nodes (retriever, reader ...). Maybe we can find another, cleaner workaround :thinking: